### PR TITLE
Add REST API docs

### LINKED ISSUE #3314: Add some documentation for REST API
Ideally, we should also make sure that very common use cases for the REST API are also covered via the UI (e.g., download all explorative tracings or similar).

### PR DESCRIPTION
**Pull request by fm3**
Should be merged only after #3385 .

https://docs.webknossos.org/v/doc-rest-api/reference/rest_api

### Issues:
- fixes #3314 

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- [x] Updated [documentation](../blob/master/docs) if applicable
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

File: docs/rest_api.md [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+
+---
+### `GET /users`
+
+List all users for which you have read access
+
+#### Expects
+ - Optional GET parameter `isEditable=[BOOLEAN]`
+   - If true: list only users for which you can change team memberships
+   - If false: list only users for which you can not change team memberships
```
Comment (jfrohnhofen) [inline docs/rest_api.md]: cannot
```suggestion
   - If false: list only users for which you cannot change team memberships
```

File: docs/rest_api.md [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+
+---
+### `GET /datasets`
+
+List all datasets for which you have read access
+
+#### Expects
+ - Optional GET parameter `isEditable=[BOOLEAN]`
+   - If true: list only datasets you can edit
+   - If false: list only datasets you can not edit
```
Comment (jfrohnhofen) [inline docs/rest_api.md]:
```suggestion
   - If false: list only datasets you cannot edit
```

File: docs/rest_api.md [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+   - `"description"` (optional) `[STRING]`
+
+
+#### Returns
+
+ - JSON list with status reports about the created tasks, including their ids
+
+#### Note
+
+For each attached NML file, one task is created. This request will also return status code 200 if some tasks could not be created. Failed tasks are instead encoded in the JSON response
```
Comment (jfrohnhofen) [inline docs/rest_api.md]: I don't get this. Above it says "Create tasks without attached NML files" but here we talk about NMLs?

File: docs/rest_api.md [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+---
+### `POST /tasks/list`
+
+List tasks matching search criteria
+
+#### Expects
+ - JSON object with four optional fields:
+   - `"user"` `[STRING]` show only tasks on which the user with this id has worked
+   - `"project"` `[STRING]` show only tasks of the project with this name
+   - `"ids"` `[JSON LIST OF STRINGS]` show only tasks with these ids
```
Comment (jfrohnhofen) [inline docs/rest_api.md]: Inconsistent with `LIST[STRING]` used above

**Comment by fm3** [comment]
note to self: also reduce indent of second point in task createFromFiles

File: docs/rest_api.md (hunk 4 above)
Comment (fm3) [reply docs/rest_api.md]: (changed the one above)

**Pull request merged**